ui/src/utils/buildQueriesForLayouts.ts

```diff
@@ -48,30 +52,52 @@ const buildCannedDashboardQuery = (
   return text
 }
 
+const addTimeBoundsToRawText = (rawText: string): string => {
+  if (!rawText) {
+    return
+  }
+
+  const dashboardTimeText: string = `time > ${TEMP_VAR_DASHBOARD_TIME}`
```

Comment:
making the string search a regex could be good.

Reply:
If `time > othertimevalue`, it should not add the upper time constraint because its a custom time bound. If `time > :dashboard` is found, it is assumed that the query is using the time selector as the bounds; this is the case that it should try and add the upper time constraint.

I will change the strings that are used for matching to regex.